matplotlib-stubs/pyplot.pyi

```diff
@@ -113,4 +115,38 @@ def subplots_adjust(
 ) -> None: ...
 def close(fig: Union[Figure, Literal["all"]]) -> None: ...
 def clf() -> None: ...
+def plot(*args: Any, scalex: bool = ..., scaley: bool = ..., data: Optional[_Data] = ..., **kwargs: Any) -> Line2D:
 def show() -> None: ...
+def xlim(*args: Any, **kwargs: Any) -> Tuple[float, float]: ...
+def ylim(*args: Any, **kwargs: Any) -> Tuple[float, float]: ...
+def xticks(
+    ticks: Optional[List[float]] = ..., labels: Optional[List[str]] = ..., **kwargs
+) -> Tuple[List[float], List[str]]: ...
+def yticks(
+    ticks: Optional[List[float]] = ..., labels: Optional[List[str]] = ..., **kwargs
+) -> Tuple[List[float], List[str]]: ...
+def xlabel(
+    ylabel: str,
+    fontdict: Optional[Dict[str, Union[str, int]]] = ...,
+    labelpad: Optional[float] = ...,
+    **kwargs: Any,
+) -> str: ...
+def ylabel(
+    ylabel: str,
+    fontdict: Optional[Dict[str, Union[str, int]]] = ...,
+    labelpad: Optional[float] = ...,
+    **kwargs: Any,
+) -> str: ...
+def fill_between(
+    x: Union[list, _np.ndarray],
+    y1: Union[list, _np.ndarray],
+    y2: Optional[Union[list, _np.ndarray]] = ...,
+    where: Optional[List[bool]] = ...,
+    interpolate: bool = ...,
+    step: Optional[Literal["pre", "post", "mid"]] = ...,
+    *,
+    data: Optional[_Data] = ...,
+    **kwargs: Any,
+) -> PolyCollection: ...
+def axhline(y: Optional[float] = ..., xmin: Optional[float] = ..., xmax: Optional[float] = ...) -> Line2d: ...
+def axvline(x: Optional[float] = ..., ymin: Optional[float] = ..., ymax: Optional[float] = ...) -> Line2d: ...
```

Comment:
Are you sure these can all be `None`?

Reply:
According to the source code, all the arguments, strangely enough, are Optional.

Reply:
👍

Reply:
I think it's "optional" in the same sense as the numpy docs say args with default values are "optional" (based on the below which is from [here](https://matplotlib.org/3.1.1/_modules/matplotlib/pyplot.html#axvline))
`def axvline(x=0, ymin=0, ymax=1, **kwargs)`

Reply:
Ah I see, yes rather than the source code I meant the docs.

Reply:
Trying to disentangle the various definitions of array-like and optional is a real pain.